java-security-test/src/main/java/com/sap/cloud/security/test/SecurityTest.java

```diff
@@ -0,0 +1,348 @@
+package com.sap.cloud.security.test;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.sap.cloud.security.config.Service;
+import com.sap.cloud.security.test.jetty.JettyTokenAuthenticator;
+import com.sap.cloud.security.token.Token;
+import com.sap.cloud.security.token.TokenClaims;
+import com.sap.cloud.security.token.TokenHeader;
+import com.sap.cloud.security.xsuaa.client.OAuth2ServiceEndpointsProvider;
+import com.sap.cloud.security.xsuaa.client.OAuth2TokenServiceConstants;
+import com.sap.cloud.security.xsuaa.client.XsuaaDefaultEndpoints;
+import org.apache.commons.io.IOUtils;
+import org.eclipse.jetty.annotations.AnnotationConfiguration;
+import org.eclipse.jetty.plus.webapp.EnvConfiguration;
+import org.eclipse.jetty.plus.webapp.PlusConfiguration;
+import org.eclipse.jetty.security.ConstraintSecurityHandler;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.servlet.FilterHolder;
+import org.eclipse.jetty.servlet.ServletHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.webapp.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.servlet.DispatcherType;
+import javax.servlet.Filter;
+import javax.servlet.Servlet;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.util.*;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+import static com.sap.cloud.security.xsuaa.client.OidcConfigurationService.DISCOVERY_ENDPOINT_DEFAULT;
+
+public class SecurityTest {
+
+    protected static final Logger LOGGER = LoggerFactory.getLogger(SecurityTest.class);
+
+    // DEFAULTS
+    public static final String DEFAULT_APP_ID = "xsapp!t0815";
+    public static final String DEFAULT_CLIENT_ID = "sb-clientId!t0815";
+    public static final String DEFAULT_DOMAIN = "localhost";
+
+    protected static final String LOCALHOST_PATTERN = "http://localhost:%d";
+
+    protected final Map<String, ServletHolder> applicationServletsByPath = new HashMap<>();
+    protected final List<FilterHolder> applicationServletFilters = new ArrayList<>();
+    // app server
+    protected Server applicationServer;
+    protected ApplicationServerOptions applicationServerOptions;
+    protected boolean useApplicationServer;
+
+    // mock server
+    protected WireMockServer wireMockServer;
+    protected RSAKeys keys;
+    protected int wireMockPort = 0;
```

Comment:
I think we could get rid of that instance variable now because it is not really needed anymore.

Reply:
should be done